src-plugins/itkFilters/itkMorphologicalFiltersProcessBase_p.h

```diff
@@ -48,8 +56,94 @@ class ITKFILTERSPLUGIN_EXPORT itkMorphologicalFiltersProcessBasePrivate : public
         }
     }
 
+
+    template <class PixelType> void update ( void )
+    {
+        typedef itk::Image< PixelType, 3 > ImageType;
+
+        if(!isRadiusInPixels)
+        {
+            convertMmInPixels<ImageType>();
+        }
+
+        typedef itk::FlatStructuringElement < 3> StructuringElementType;
+        StructuringElementType::RadiusType elementRadius;
+        elementRadius[0] = radius[0]; //radius (double) is truncated
+        elementRadius[1] = radius[1];
+        elementRadius[2] = radius[2];
+        StructuringElementType ball = StructuringElementType::Ball(elementRadius);
+
+        typedef itk::MinimumMaximumImageFilter <ImageType> ImageCalculatorFilterType;
+        typename ImageCalculatorFilterType::Pointer imageCalculatorFilter = ImageCalculatorFilterType::New();
+        imageCalculatorFilter->SetInput( dynamic_cast<ImageType *> ( ( itk::Object* ) ( input->data() ) ) );
+        imageCalculatorFilter->Update();
+
+        typedef itk::KernelImageFilter< ImageType, ImageType, StructuringElementType >  FilterType;
+        typedef itk::BinaryDilateImageFilter< ImageType, ImageType,StructuringElementType >  DilateFilterType;
+        typedef itk::BinaryErodeImageFilter< ImageType, ImageType,StructuringElementType >  ErodeFilterType;
+        typedef itk::BinaryMorphologicalClosingImageFilter< ImageType, ImageType, StructuringElementType >  CloseFilterType;
+        typedef itk::BinaryMorphologicalOpeningImageFilter< ImageType, ImageType, StructuringElementType >  OpenFilterType;
+        typename FilterType::Pointer filter;
+
+        if(description == "Dilate filter")
```

Comment:
Since you are already doing a dynamic cast to the specific filter type, why not use that cast as a test of the filter type ?
```
FooFilterType fooFilter = dynamic_cast<FooFilterType*>(...)
if (fooFilter)
{
   blah blah
}
etc.